framework/logging.cpp

```diff
@@ -1571,7 +1653,13 @@ static int print_one_thread_messages(int fd, struct per_thread_data *data, struc
             if (!newline) {
                 /* single line */
                 char c = '\'';  // I'm lazy
-                print_content_single_line(fd, "   - '", message, std::string_view(&c, 1));
+                if(message[0]=='S'){
+                    std::string message_string=std::string(message);
+                    std::string skip_message=message_string.substr(0,3)+message_string.substr(4,message_string.size());
+                    std::string_view skip_message_view(&skip_message[0],skip_message.size());
+                    print_content_single_line(fd, "   - '", skip_message_view, std::string_view(&c, 1));
+                }else
+                    print_content_single_line(fd, "   - '", message, std::string_view(&c, 1));
```

Comment:
This part calls for some comments. Why 'S'? I'm assuming that's how skipped tests' log message begins (with `S>`). Which characters are the `substr` meant to remove?

Reply:
the skip messages always start with `S> ` but there is a skip category that the framework needs to encode along with the skip message. For e.g., if the skip category is `RESOURCE_UNAVAILABLE` which corresponds to letter `A`, this needs to be part of the skip message shared to the main sandstone process. When decoding on the main process side, we need to exclude it when logging. 

For e.g.,

Encoded message: `S> ATest skipped due to resource unavailability`
Decode message:   `S> Test skipped due to resource unavailability` No `A` here.

Reply:
The skip message could be another entry in the log type enum:
https://github.com/opendcdiag/opendcdiag/blob/2ac0d930ae7f3472794d4e05fbfdfd014c928e44/framework/logging.cpp#L109-L113